packager/packager.cc

```diff
@@ -1560,71 +1560,91 @@ unique_ptr<PackageInfoImpl> createAndPopulatePackageInfo(core::GlobalState &gs,
 // https://www.cs.cmu.edu/~15451-f18/lectures/lec19-DFS-strong-components.pdf provides a good overview of the
 // algorithm.
 const int UNVISITED = 0;
-struct ComputeSCCsMetadata {
-    int nextIndex = 1;
-    int nextSCCId = 0;
+struct NodeInfo {
     // A given package's index in the DFS traversal; ie. when it was first visited. The default value of 0 means the
     // package hasn't been visited yet.
-    UnorderedMap<core::packages::MangledName, int> index;
+    int index = UNVISITED;
     // The lowest index reachable from a given package (in the same SCC) by following any number of tree edges
     // and at most one back/cross edge
-    UnorderedMap<core::packages::MangledName, int> lowLink;
+    int lowLink = 0;
     // Fast way to check if a package is on the stack
-    UnorderedMap<core::packages::MangledName, bool> onStack;
+    bool onStack = false;
+};
+
+struct ComputeSCCsMetadata {
+    int nextIndex = 1;
+    int nextSCCId = 0;
+    UnorderedMap<core::packages::MangledName, NodeInfo> nodeMap;
     // As we visit packages, we push them onto the stack. Once we find the "root" of an SCC, we can use the stack to
     // determine all packages in the SCC.
     std::vector<core::packages::MangledName> stack;
 };
 
 // DFS traversal for Tarjan's algorithm starting from pkgName, along with keeping track of some metadata needed for
 // detecting SCCs.
-void strongConnect(core::GlobalState &gs, ComputeSCCsMetadata &metadata, core::packages::MangledName pkgName) {
+void strongConnect(core::GlobalState &gs, ComputeSCCsMetadata &metadata, core::packages::MangledName pkgName,
+                   NodeInfo &infoAtEntry) {
     auto *pkgInfoPtr = gs.packageDB().getPackageInfoNonConst(pkgName);
     if (!pkgInfoPtr) {
         // This is to handle the case where the user imports a package that doesn't exist.
         return;
     }
     auto &pkgInfo = PackageInfoImpl::from(*pkgInfoPtr);
-    metadata.index[pkgName] = metadata.nextIndex;
-    metadata.lowLink[pkgName] = metadata.nextIndex;
-    metadata.nextIndex++;
-    metadata.stack.push_back(pkgName);
-    metadata.onStack[pkgName] = true;
+    {
+        infoAtEntry.index = metadata.nextIndex;
+        infoAtEntry.lowLink = metadata.nextIndex;
+        metadata.nextIndex++;
+        metadata.stack.push_back(pkgName);
+        infoAtEntry.onStack = true;
+    }
```

Comment:
Why did this end up in a new block?

Reply:
One of the previous commits did:

```c++
  auto &info = metadata.nodeMap[pkgName];
```

and I thought it was best to put that in its own block to limit the lifetime of `info`.  I can take it out now that it's served its purpose.